modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_items.dm

```diff
@@ -0,0 +1,143 @@
+/obj/item/soulcatcher_item
+	name = "'Evoker'-Type RSD"
+	desc = "The 'Evoker'-Type RSD, or Resonance Simulation Device, is a sort of 'Soulcatcher' instrument that's been designated for handheld usage. These RSDs were designed with the Medical field in mind, a tool meant to offer comfort to the temporarily-departed while their bodies are being repaired, healed, or produced. The Evoker is essentially a very specialized handheld NIF, still using the same nanomachinery for the software and hardware. This careful instrument is able to host a virtual space for a great number of Engrams for an essentially indefinite amount of time in an unlimited variety of simulations, even able to transfer them to and from a NIF. However, it's best Medical practice to not lollygag."
+	icon = 'modular_skyrat/modules/modular_implants/icons/obj/devices.dmi'
+	icon_state = "soulcatcher-device"
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	/// What soulcatcher datum is associated with this item?
+	var/datum/component/soulcatcher/linked_soulcatcher
+
+/obj/item/soulcatcher_item/attack_self(mob/user, modifiers)
+	linked_soulcatcher.ui_interact(user)
+
+/obj/item/soulcatcher_item/New(loc, ...)
+	. = ..()
+	linked_soulcatcher = AddComponent(/datum/component/soulcatcher)
+
+/obj/item/soulcatcher_item/Destroy(force)
+	if(linked_soulcatcher)
+		qdel(linked_soulcatcher)
+
+	return ..()
+
+/obj/item/soulcatcher_item/attack(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
```

Comment:
Can you please make sure that you can't use it at a range, or that if we're waiting on a ghost's input, that it gets cancelled or something if you move away (or that it just doesn't continue, even if they don't accept or something)?

Reply:
I just tested it, it does not indeed work at a range.

Reply:
This has been tested and it does not function at a range.